### DISCUSSION TITLE
A common pool reported dRAID and must use 16k volsize

### DISCUSSION THREAD

**Discussion by guanfeix** [openzfs/zfs#16651] [General]
environment
![image](https://github.com/user-attachments/assets/6eb270b1-e27f-4c8b-9d6f-1c93f9b57181)

pool

![image](https://github.com/user-attachments/assets/35d7aa59-d593-434e-8ee8-cd56d3095ff5)
![image](https://github.com/user-attachments/assets/f2dddc7c-b222-4d15-aa36-e9a6e2e64d0e)

disk detail:

![image](https://github.com/user-attachments/assets/14559dc4-6b88-48e0-890a-65afb14e8b46)
![image](https://github.com/user-attachments/assets/cc1c8ce7-11f8-46b1-ba7b-889e0519865a)

rdfs is a alias name 
![image](https://github.com/user-attachments/assets/96b3d905-edf2-4fce-9206-136d1b0986cf)

**Comment by guanfeix** [comment]
ZFS Version: 2.1.15

**Comment by guanfeix** [comment]
Pool create cmd:
![image](https://github.com/user-attachments/assets/4fa0a63c-d122-41c0-8a1c-b4d2d9dadaa7)

**Comment by guanfeix** [comment]
question:
Warning: volblocksize (8192) is much less than the minimum allocation
unit (16384)，which wastes at least 50% of space. To reduce wasted space,
use a larger volblocksize (16384 is recommended),fewer dRAID data disks
per group, or smaller sector size (ashift).

there is no such problem for some other pool

**Comment by amotin** [comment]
First, as I see the pool is not reported as dRAID, the message is merely meaning that if you use dRAID, then you may think about reducing the number of data disks per group.  Second, RAIDZ2 is indeed very inefficient for storing small objects, especially of disks with 4KB physical sector size.  To store up to 4KB of data RAIDZ2 would need 12KB of space, to store up to 16KB of data it would need 24KB of space, to store up to 28KB of data -- 36KB of space, etc.  You are creating ZVOL in which every block will have size of no more than 8KB, which means you will likely use 2x more space than required and have pretty low sequential performance.  So either increase the block size, or, if you really don't want for some very good reason, use mirror (tripple mirror if your really need that redundancy).  There is no bug here.
